Build failure for planck_rev6 with KSCAN_MATRIX_POLLING

Steps to reproduce: 

1. Switch to latest commit a7e8655287e93ce16c14c191dd8e6eca13fa48db
1. Add `CONFIG_ZMK_KSCAN_MATRIX_POLLING=y` to `app/boards/arm/planck/planck_rev6_defconfig`
1. Run `west build -p -b planck_rev6`

Error message:
```
[131/158] Building C object modules/drivers/CMakeFiles/..__app__drivers__zephyr.dir/kscan_gpio_matrix.c.obj
FAILED: modules/drivers/CMakeFiles/..__app__drivers__zephyr.dir/kscan_gpio_matrix.c.obj 
ccache /home/nuxiom/zephyr-sdk-0.11.4/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DBUILD_VERSION=zephyr-v2.2.0-3207-g2857037bdd3b -DCORE_CM4 -DHSE_VALUE=8000000 -DKERNEL -DSTM32F303xC -DUSE_FULL_LL_DRIVER -DUSE_HAL_DRIVER -D_FORTIFY_SOURCE=2 -D__PROGRAM_START -D__ZEPHYR__=1 -I/home/nuxiom/zmk/zephyr/include -Izephyr/include/generated -I/home/nuxiom/zmk/zephyr/soc/arm/st_stm32/stm32f3 -I/home/nuxiom/zmk/zephyr/drivers -I/home/nuxiom/zmk/zephyr/subsys/usb -I/home/nuxiom/zmk/modules/hal/cmsis/CMSIS/Core/Include -I/home/nuxiom/zmk/modules/hal/stm32/stm32cube/stm32f3xx/soc -I/home/nuxiom/zmk/modules/hal/stm32/stm32cube/stm32f3xx/drivers/include -I/home/nuxiom/zmk/modules/hal/stm32/stm32cube/stm32f3xx/drivers/include/Legacy -I../drivers/zephyr/. -isystem /home/nuxiom/zmk/zephyr/lib/libc/minimal/include -isystem /home/nuxiom/zephyr-sdk-0.11.4/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/include -isystem /home/nuxiom/zephyr-sdk-0.11.4/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/include-fixed -Os -imacros/home/nuxiom/zmk/app/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -mcpu=cortex-m4 -mthumb -mabi=aapcs -imacros/home/nuxiom/zmk/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-address-of-packed-member -Wno-pointer-sign -Wpointer-arith -Wno-unused-but-set-variable -Werror=implicit-int -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/home/nuxiom/zmk/app=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/nuxiom/zmk/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/nuxiom/zmk=WEST_TOPDIR -ffunction-sections -fdata-sections -std=c99 -nostdinc -Wfatal-errors -MD -MT modules/drivers/CMakeFiles/..__app__drivers__zephyr.dir/kscan_gpio_matrix.c.obj -MF modules/drivers/CMakeFiles/..__app__drivers__zephyr.dir/kscan_gpio_matrix.c.obj.d -o modules/drivers/CMakeFiles/..__app__drivers__zephyr.dir/kscan_gpio_matrix.c.obj -c ../drivers/zephyr/kscan_gpio_matrix.c
In file included from /home/nuxiom/zmk/zephyr/include/sched_priq.h:9,
                 from /home/nuxiom/zmk/zephyr/include/kernel_includes.h:23,
                 from /home/nuxiom/zmk/zephyr/include/kernel.h:17,
                 from /home/nuxiom/zmk/zephyr/include/init.h:11,
                 from /home/nuxiom/zmk/zephyr/include/device.h:22,
                 from ../drivers/zephyr/kscan_gpio_matrix.c:9:
../drivers/zephyr/kscan_gpio_matrix.c: In function 'kscan_gpio_init_0':
../drivers/zephyr/kscan_gpio_matrix.c:250:32: error: 'kscan_gpio_irq_callback_handler_0' undeclared (first use in this function); did you mean 'kscan_gpio_irq_callback_0'?
  250 |                                kscan_gpio_irq_callback_handler_##n, BIT(in_cfg->pin));             \
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/nuxiom/zmk/zephyr/include/sys/util.h:447:26: note: in definition of macro '__DEBRACKET'
  447 | #define __DEBRACKET(...) __VA_ARGS__
      |                          ^~~~~~~~~~~
/home/nuxiom/zmk/zephyr/include/sys/util.h:444:2: note: in expansion of macro '__GET_ARG2_DEBRACKET'
  444 |  __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |  ^~~~~~~~~~~~~~~~~~~~
/home/nuxiom/zmk/zephyr/include/sys/util.h:327:2: note: in expansion of macro '__COND_CODE'
  327 |  __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |  ^~~~~~~~~~~
/home/nuxiom/zmk/zephyr/include/sys/util.h:324:2: note: in expansion of macro 'Z_COND_CODE_1'
  324 |  Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |  ^~~~~~~~~~~~~
/home/nuxiom/zmk/zephyr/include/devicetree.h:1698:2: note: in expansion of macro 'COND_CODE_1'
 1698 |  COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT), \
      |  ^~~~~~~~~~~
zephyr/include/generated/devicetree_unfixed.h:4817:56: note: in expansion of macro 'GPIO_INST_INIT'
 4817 | #define DT_FOREACH_OKAY_INST_zmk_kscan_gpio_matrix(fn) fn(0)
      |                                                        ^~
/home/nuxiom/zmk/zephyr/include/sys/util.h:510:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_zmk_kscan_gpio_matrix'
  510 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
../drivers/zephyr/kscan_gpio_matrix.c:293:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'
  293 | DT_INST_FOREACH_STATUS_OKAY(GPIO_INST_INIT)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
[140/158] Building C object zephyr/kernel/CMakeFiles/kernel.dir/sched.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /home/nuxiom/zmk/app/build
```

Also this error does not occur when building without explicitly adding `CONFIG_ZMK_KSCAN_MATRIX_POLLING=y` to `app/boards/arm/planck/planck_rev6_defconfig`. Which seems to suggest that the default config in `app/boards/arm/planck/Kconfig.defconfig` did not apply.

--- a/app/drivers/zephyr/kscan_gpio_matrix.c
+++ b/app/drivers/zephyr/kscan_gpio_matrix.c
@@ -181,19 +181,19 @@ static int kscan_gpio_config_interrupts(struct device **devices,
         struct kscan_gpio_data_##n *data = CONTAINER_OF(work, struct kscan_gpio_data_##n, work);   \
         kscan_gpio_read_##n(data->dev);                                                            \
     }                                                                                              \
-    COND_CODE_1(CONFIG_ZMK_KSCAN_MATRIX_POLLING, (),                                               \
-                (static void kscan_gpio_irq_callback_handler_##n(                                  \
-                    struct device *dev, struct gpio_callback *cb, gpio_port_pins_t pin) {          \
-                    struct kscan_gpio_irq_callback_##n *data =                                     \
-                        CONTAINER_OF(cb, struct kscan_gpio_irq_callback_##n, callback);            \
-                    kscan_gpio_disable_interrupts_##n(data->dev);                                  \
-                    COND_CODE_0(DT_INST_PROP(n, debounce_period),                                  \
-                                ({ k_work_submit(data->work); }), ({                               \
-                                    k_delayed_work_cancel(data->work);                             \
-                                    k_delayed_work_submit(                                         \
-                                        data->work, K_MSEC(DT_INST_PROP(n, debounce_period)));     \
-                                }))                                                                \
-                }))                                                                                \
+    static void kscan_gpio_irq_callback_handler_##n(                                               \
+        struct device *dev, struct gpio_callback *cb, gpio_port_pins_t pin) {                      \
+        struct kscan_gpio_irq_callback_##n *data =                                                 \
+            CONTAINER_OF(cb, struct kscan_gpio_irq_callback_##n, callback);                        \
+        COND_CODE_1(CONFIG_ZMK_KSCAN_MATRIX_POLLING, (),                                           \
+                    (kscan_gpio_disable_interrupts_##n(data->dev);))                               \
+        COND_CODE_0(DT_INST_PROP(n, debounce_period),                                              \
+                    ({ k_work_submit(data->work); }), ({                                           \
+                        k_delayed_work_cancel(data->work);                                         \
+                        k_delayed_work_submit(                                                     \
+                            data->work, K_MSEC(DT_INST_PROP(n, debounce_period)));                 \
+                    }))                                                                            \
+    }                                                                                              \
                                                                                                    \
     static struct kscan_gpio_data_##n kscan_gpio_data_##n = {                                      \
         .rows = {[INST_MATRIX_ROWS(n) - 1] = NULL}, .cols = {[INST_MATRIX_COLS(n) - 1] = NULL}};   \

--- a/app/drivers/zephyr/kscan_gpio_matrix.c
+++ b/app/drivers/zephyr/kscan_gpio_matrix.c
@@ -181,17 +181,16 @@ static int kscan_gpio_config_interrupts(struct device **devices,
         struct kscan_gpio_data_##n *data = CONTAINER_OF(work, struct kscan_gpio_data_##n, work);   \
         kscan_gpio_read_##n(data->dev);                                                            \
     }                                                                                              \
-    static void kscan_gpio_irq_callback_handler_##n(                                               \
-        struct device *dev, struct gpio_callback *cb, gpio_port_pins_t pin) {                      \
+    static void kscan_gpio_irq_callback_handler_##n(struct device *dev, struct gpio_callback *cb,  \
+                                                    gpio_port_pins_t pin) {                        \
         struct kscan_gpio_irq_callback_##n *data =                                                 \
             CONTAINER_OF(cb, struct kscan_gpio_irq_callback_##n, callback);                        \
         COND_CODE_1(CONFIG_ZMK_KSCAN_MATRIX_POLLING, (),                                           \
                     (kscan_gpio_disable_interrupts_##n(data->dev);))                               \
-        COND_CODE_0(DT_INST_PROP(n, debounce_period),                                              \
-                    ({ k_work_submit(data->work); }), ({                                           \
+        COND_CODE_0(DT_INST_PROP(n, debounce_period), ({ k_work_submit(data->work); }), ({         \
                         k_delayed_work_cancel(data->work);                                         \
-                        k_delayed_work_submit(                                                     \
-                            data->work, K_MSEC(DT_INST_PROP(n, debounce_period)));                 \
+                        k_delayed_work_submit(data->work,                                          \
+                                              K_MSEC(DT_INST_PROP(n, debounce_period)));           \
                     }))                                                                            \
     }                                                                                              \
                                                                                                    \

